diff --git a/src/Makefile.am b/src/Makefile.am
index 794e668..7a73db6 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -32,10 +32,10 @@ endif
 AM_CXXFLAGS += $(GLIB_DEPS_CFLAGS)
 noinst_PROGRAMS += calfmakerdf
 calfmakerdf_SOURCES = makerdf.cpp
-calfmakerdf_LDADD = calf.la
+calfmakerdf_LDADD = calf.la $(JACK_DEPS_LIBS)
 
 calfbenchmark_SOURCES = benchmark.cpp
-calfbenchmark_LDADD = calf.la
+calfbenchmark_LDADD = calf.la $(JACK_DEPS_LIBS)
 
 calf_la_SOURCES = audio_fx.cpp analyzer.cpp lv2wrap.cpp metadata.cpp modules_tools.cpp modules_delay.cpp modules_comp.cpp modules_limit.cpp modules_dist.cpp modules_filter.cpp modules_mod.cpp modules_pitch.cpp fluidsynth.cpp giface.cpp monosynth.cpp organ.cpp osctl.cpp plugin.cpp preset.cpp synth.cpp utils.cpp wavetable.cpp modmatrix.cpp
 calf_la_LIBADD = $(FLUIDSYNTH_DEPS_LIBS) $(GLIB_DEPS_LIBS) 
